clientconn.go

```diff
@@ -361,11 +361,11 @@ func (cc *ClientConn) lbWatcher() {
 
 func (cc *ClientConn) newAddrConn(addr Address, skipWait bool) error {
 	ac := &addrConn{
-		cc:           cc,
-		addr:         addr,
-		dopts:        cc.dopts,
-		shutdownChan: make(chan struct{}),
+		cc:    cc,
+		addr:  addr,
+		dopts: cc.dopts,
 	}
+	ac.dopts.copts.Cancel = make(chan struct{})
```

Comment:
Removing shutdownChan seems hurt the code readability in my opinion -- letting addrConn work with a transport layer connection option sounds strange when reading the code. I would like to keep shutdownChan and assign it to ac.dopts.copts.Cancel before we call transport.NewClientTransport(...). The point is to operate on connection options only when we want to create a transport.